### PR TITLE
v3 signmessage fw parsing in trezorlib

### PR DESCRIPTION
**Pull request by matejcik**
introduces the necessary pieces for validation of v3-signed T1 firmwares

will cause a conflict with #2649 -- @TychoVrahe please remove the piece touching trezorlib because it's already included here

### CODE REVIEW

File: python/src/trezorlib/firmware/models.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    bootloader_sigs_needed=-1,
+    firmware_keys=[
+        bytes.fromhex(key)
+        for key in (
+            "037308e14077161c365dea0f5c80aa6c5dba34719e825bd23ae5f7e7d2988adb0f",
+            "039c1b2460e343712e982e0732e7ed17f60de4c933065b7170d99c6e7fe7cc7f4b",
+            "03152b37fdf126111274c894c348dcc975b57c115ee24ceb19b5190ac7f7b65173",
+        )
+    ],
+    firmware_sigs_needed=2,
```
Comment (TychoVrahe) [inline python/src/trezorlib/firmware/models.py:110]: the signing script in debug_signatures actually fill all three signatures, which causes the validation to fail. we should increase this to 3 or modify the script

**Review by TychoVrahe — CHANGES_REQUESTED** [review]
needs fixing validation of  legacy V3 DEV signatures

File: python/src/trezorlib/firmware/models.py (hunk 1 above)
Comment (TychoVrahe) [reply python/src/trezorlib/firmware/models.py:110]: will be handled elsewhere

File: python/src/trezorlib/firmware/legacy.py [hunk 2]
```diff
@@ -73,17 +104,49 @@ class LegacyV2Firmware(FirmwareImage):
...
+            public_keys = models.TREZOR_ONE_V3.firmware_keys
+        else:
+            public_keys = models.TREZOR_ONE_V3_DEV.firmware_keys
+
+        self.validate_code_hashes()
+        check_sig_signmessage(
+            self.digest(),
+            self.header.v1_key_indexes,
+            self.header.v1_signatures,
+            models.TREZOR_ONE_V3.firmware_sigs_needed,
```
[the remark this answers is not in the document]
Comment (matejcik) [reply python/src/trezorlib/firmware/legacy.py]: [e4a08a8](https://github.com/trezor/trezor-firmware/pull/2701/commits/e4a08a86ca96987d14b5751db3ab1bf417d6b391)

File: python/src/trezorlib/firmware/legacy.py [hunk 3]
```diff
@@ -126,15 +189,18 @@ class LegacyFirmware(Struct):
...
+    def verify(self, dev_keys: bool = False) -> None:
+        if not dev_keys:
+            public_keys = models.TREZOR_ONE_V1V2.firmware_keys
+        else:
+            public_keys = models.TREZOR_ONE_V1V2_DEV.firmware_keys
         check_sig_v1(
             self.digest(),
             self.key_indexes,
             self.signatures,
+            models.TREZOR_ONE_V1V2.firmware_sigs_needed,
```
[the remark this answers is not in the document]
Comment (matejcik) [reply python/src/trezorlib/firmware/legacy.py]: [e4a08a8](https://github.com/trezor/trezor-firmware/pull/2701/commits/e4a08a86ca96987d14b5751db3ab1bf417d6b391)

File: python/src/trezorlib/_internal/firmware_headers.py [hunk 4]
```diff
@@ -333,15 +379,14 @@ def format(self, verbose: bool = False) -> str:
...
         self.validate_code_hashes()
-        if not public_keys:
-            public_keys = self.public_keys()
+        public_keys = self.public_keys(dev_keys)
         try:
             cosi.verify(
                 self.header.signature,
                 self.digest(),
-                firmware.V2_SIGS_REQUIRED,
+                fw_models.TREZOR_T.boardloader_sigs_needed,
```
[the remark this answers is not in the document]
Comment (matejcik) [reply python/src/trezorlib/_internal/firmware_headers.py]: [b729b16](https://github.com/trezor/trezor-firmware/pull/2701/commits/b729b167cea17671c6faae0870150ed9174fc398)

**Pull request merged**